frontend/src/actions/WindowActions.js

```diff
@@ -312,9 +313,16 @@ export function updateDataFieldProperty(property, item, scope) {
   };
 }
 
-export function noConnection(status) {
+export function noConnection({ errorType }) {
   return {
-    type: NO_CONNECTION,
+    type: CONNECTION_ERROR,
+    errorType,
+  };
+}
+
+export function badGateway(status) {
```

Comment:
imho you shall unify those actions into ONE action: connectionError which takes a errorType param which can be:
* OFFLINE
* BAD_GATEWAY
* maybe more to be added in future

Reply:
we have the `noConnection({ errorType }` I forgot to delete this one